src/main/webapp/app/exam/participate/exercises/programming/programming-exam-submission.component.ts

```diff
@@ -114,6 +137,18 @@ export class ProgrammingExamSubmissionComponent extends ExamSubmissionComponent
         }
     }
 
+    /**
+     * Called when the build is completed after a new commit.
+     * Intended only for the case when the online editor is disabled (otherwise, we can update on CommitState changes)
+     */
+    onBuildComplete() {
+        if (this.exercise.allowOnlineEditor || !this.exercise.allowOfflineIde) {
+            return;
+        }
+        // this.setSubmissionCountAndLockIfNeeded(this.studentParticipation.submissionCount);
+        // FIXME: this.studentParticipation contains the old submissionCount value here -> can't update submissionCount
```

Comment:
Not sure how to handle this. Currently, when the code editor is disabled the only way to update `submissionCount` is to refresh the page. 

Should I increment `submissionCount` locally and compare it to `exercise.submissionPolicy.submissionLimit` to determine if the repo should be locked? 🤔

Reply:
When the student pushes code, the client should receive a websocket message with the programming submission. This should contain a link to the participation with the submission count and locked=true/false

So the server is already calculating the required information and you need to extract and update the local objects with the values from the websocket message. Can you try if this is possible? (Hopefully without too many changes)